kernel/main.rs

```diff
@@ -5,8 +5,9 @@
            unwind_attributes, zero_one, collections_range, question_mark)]
 #![no_std]
 
-#![allow(deprecated)]
-// #![deny(warnings)]
+// Its me, ur friend. If you ever comment this line, you will end up in hell. Seriously, don't. You
```

Comment:
I will not bring in this commit. There are good reasons to have warnings at the moment, as we have areas of the code like dup that are not feature complete, and the programmer should be reminded of that every time they build the kernel, but still be able to run the kernel